src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -423,5 +459,41 @@ public virtual void Shutdown()
 
             Process.GetCurrentProcess().Close();
         }
+
+        private void OnHostHealthCheckTimer(object state)
+        {
+            if (State == ScriptHostState.Running && !CheckHostHealth())
+            {
+                // This periodic check allows us to break out of the host run
+                // loop. The health check performed in OnHostStarting will then
+                // fail and we'll enter a restart loop (exponentially backing off)
+                // until the host is healthy again and we can resume host processing.
+                RestartHost();
+            }
+        }
+
+        private bool CheckHostHealth(bool throwWhenUnhealthy = false)
```

Comment:
nit: name doesn't make sense for a function that returns a bool. Should be something like IsHostHealthy(). Maybe split this into two functions - IsHostHealthy() and ThrowIfHostUnhealthy()

Reply:
Changed back to IsHostHealth (that's what I had earlier :))